### PR TITLE
[DNM] vendor, tests, Bump to ginkgo v2

### PR DESCRIPTION
**Pull request by RamLavi**
**What this PR does / why we need it**:
This PR bumps kubemacpool to ginkgo v2

**Special notes for your reviewer**:
Once this PR is merged we may need to update CNAO's KMP tier1 test [flags](https://github.com/kubevirt/cluster-network-addons-operator/blob/main/automation/check-patch.e2e-kubemacpool-functests.sh#L57).
**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: automation/check-patch.e2e-k8s.sh [hunk 1]
```diff
@@ -6,7 +6,7 @@ teardown() {
 }
 
 main() {
-    export KUBEVIRT_PROVIDER='k8s-1.23'
+    export KUBEVIRT_PROVIDER='k8s-1.21'
```
Comment (oshoval) [inline automation/check-patch.e2e-k8s.sh:9]: not related right ?

Comment (RamLavi) [reply automation/check-patch.e2e-k8s.sh:9]: right, thanks!

**Comment by RamLavi** [comment]
@qinqon @oshoval I'm getting a strange error 
```
package github.com/k8snetworkplumbingwg/kubemacpool: build constraints exclude all Go files in /home/ralavi/go/src/github.com/k8snetworkplumbingwg/kubemacpool
```

did we add any build constraints recently? I didn't find any on any non vendor folders.. could we be blocked by vendor folders?

**Comment by oshoval** [comment]
> @qinqon @oshoval I'm getting a strange error
> 
> ```
> package github.com/k8snetworkplumbingwg/kubemacpool: build constraints exclude all Go files in /home/ralavi/go/src/github.com/k8snetworkplumbingwg/kubemacpool
> ```
> 
> did we add any build constraints recently? I didn't find any on any non vendor folders.. could we be blocked by vendor folders?

dont know by heart
i would check GOARCH / GOOS
https://stackoverflow.com/questions/55348458/build-constraints-exclude-all-go-files-in
or this one https://github.com/golang/go/issues/24068 that also have the mentioned error

File: automation/check-patch.e2e-k8s.sh [hunk 2]
```diff
@@ -17,7 +17,7 @@ main() {
     make cluster-up
     trap teardown EXIT SIGINT SIGTERM SIGSTOP
     make cluster-sync
-    make E2E_TEST_ARGS="-ginkgo.noColor --junit-output=$ARTIFACTS/junit.functest.xml" functest
+    make E2E_TEST_ARGS="-ginkgo.noColor --ginkgo.junit-report=$ARTIFACTS/junit.functest.xml" functest
```
Comment (oshoval) [inline automation/check-patch.e2e-k8s.sh:20]: you need to use --output-dir $ARTIFACTS, and to remove the dir from the junit report please

remove also all the teardown that manipulate those files

note that spyglass doesn't print errors good until there will be a bump of it

Comment (RamLavi) [reply automation/check-patch.e2e-k8s.sh]: Let's see if [it](https://github.com/k8snetworkplumbingwg/kubemacpool/pull/362/commits/114302fa9e8229b186b1fa847b3df721417b908b) works

Comment (oshoval) [reply automation/check-patch.e2e-k8s.sh]: Thanks
Q: Dont we still need `--junit-output` in addition ?
unless the default file name that is used is good

Comment (RamLavi) [reply automation/check-patch.e2e-k8s.sh]: it isn't, you're right

File: automation/check-patch.unit-test.sh [hunk 3]
```diff
@@ -1,9 +1,5 @@
 #!/bin/bash -e
 
-teardown() {
```
Comment (oshoval) [inline automation/check-patch.unit-test.sh:3]: remove line 7 please then

**Comment by oshoval** [comment]
If the vendor folder can be in a different commit please it would be great, thanks

File: automation/check-patch.e2e-k8s.sh (hunk 2 above)
Comment (oshoval) [reply automation/check-patch.e2e-k8s.sh]: https://onsi.github.io/ginkgo/MIGRATING_TO_V2#generating-machine-readable-reports
yes
you can see other repos like CNAO i think, where it is used when bumped to ginkgo 2.0

if you want, lets postphone it to a new PR

Comment (RamLavi) [reply automation/check-patch.e2e-k8s.sh]: we do.. so.. I don't understand why it doesn't work. checking..

**Comment by RamLavi** [comment]
closing in favor of https://github.com/k8snetworkplumbingwg/kubemacpool/pull/382

**Pull request closed without merging**